tools/autograd/load_derivatives.py

```diff
@@ -88,6 +90,12 @@ def create_derivative(f: NativeFunction, formula: str, var_names: Tuple[str, ...
     formula, saved_inputs = saved_variables(formula, arguments, var_names)
     formula, saved_outputs = saved_variables(formula, named_returns, var_names)
 
+    named_grads: Dict[str, int] = {}
+    for i, return_name in enumerate(return_names):
```

Comment:
Do we need to use dictionaries here (and the associated dictionary merging logic)? The indices are already implicit in the ordering of the return names tuple.

Reply:
We shouldn't need to, but I want to be defensive against possible coding errors. We will get a compiler error if we have two entries with the same name (since it generates a local variable named such), but if we have multiple names aliasing the same index we'd only have a potential runtime error.

Reply:
hey are you suggesting instead that we just collect the list of `grad_X` names and then later on look up the corresponding index for each one right before we write to the code gen? If that's the case, then yes, we don't need to have a dict here.

It would still be good, IMO, as a final sanity check at the end to confirm that names and indexes are in 1:1 correspondence.

Reply:
OK, I think I figured out a solution that satisfies all concerns. I'll push it tonight.

Reply:
OK so now once per native function, we generate a list of possible named gradients from the list of named outputs. This list is in the same order as the outputs.

Then we use this list to find which names are used in each derivative, collecting that into a set.

Then, during codegen, we use `list.index` to identify the ID used for the alias.